src/kit/ResponsiveGroup.tsx

```diff
@@ -55,11 +55,12 @@ const ResponsiveGroup: React.FC<Props> = ({
       },
       { accumulatedWidth: 0, visible: 0 },
     );
-    setNumVisible((prev) => {
-      if (prev !== visible) onChange?.(visible);
-      return visible;
-    });
-  }, [childrenArray, gap, maxVisible, onChange, size.width]);
+
+    if (visible !== numVisible) {
+      onChange?.(visible);
+      setNumVisible(visible);
+    }
```

Comment:
nit: this does get rid of the error message, but I worry that using both `numVisible` and `setNumVisible` in the same effect will cause it to be run too often, even if it's not an endless loop. see if something like this works:
```
let visibilityChanged = false;
setNumVisible((prevVisibility) => {
   if (prevVisibility !== visible) visibilityChanged = true;
   return visible;
}
if (visibilityChanged) onChange?.(visible)
```